04-onion-routing.md

```diff
@@ -755,59 +763,153 @@ Note that these error messages are not reliable, as they are not placed on-chain
 due to the possibility of hop failure.
 
 Intermediate hops store the shared secret from the forward path and reuse it to
-obfuscate any corresponding return packet during each hop.
+authenticate and obfuscate any corresponding return packet during each hop.
 In addition, each node locally stores data regarding its own sending peer in the
 route, so it knows where to return-forward any eventual return packets.
-The node generating the error message (_erring node_) builds a return packet
+
+The origin node signals to each node on the path that it supports the error
+format that is detailed below by setting the `fat_error` tlv record in the
+forward packet to contain the desired structure. It must make sure that each
+node supports this format by observing the node feature `option_fat_error`.
+
+By choosing a structure for the fat errors, the origin node is making a
+trade-off between failure message size and privacy. For maximum privacy,
+`max_hops` should be set to the global maximum of 27. A lower value reveals a
+lower upper bound for the route length.
+
+## Erring node
+
+The node generating the error message builds a return packet
 consisting of the following fields:
 
 1. data:
-   * [`32*byte`:`hmac`]
    * [`u16`:`failure_len`]
    * [`failure_len*byte`:`failuremsg`]
    * [`u16`:`pad_len`]
    * [`pad_len*byte`:`pad`]
+   * [`max_hops*(max_payload_size+1)*byte`:`payloads`]
```

Comment:
Any reason that the 1 byte shouldn't be included in a `max_payload_size` param? Feels like a on off-by-one bug waiting to happen.

Reply:
I excluded it because I thought the 1 byte is 'internal' and needed to do error attribution, where as the other bytes are 'application specific' - as seen from lowest level.